pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/amp_to_pwa.md

```diff
@@ -0,0 +1,204 @@
+---
+
+Progressive Web Apps harness the power of [service workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) to enable rich offline abilities and consistent user experiences across varying network strengths. 
+By caching resources within the browser, a PWA is able to provide data, assets, and offline pages to the user to keep them engaged and informed. 
+
+This tutorial will teach you how to turn an AMP site into an installable PWA with offline capabilities by adding a Web Manifest and a Service Worker powered by the AMP Service Worker.
+
+
+# Download and run the starter code 
+
+Download the [starter code here](../../../../../static/files/tutorials/amptopwa.zip).
+
+Use a local web server to preview the website.
+
+[tip type="default"]
+**TIP –**  For a quick web server, run `python -m SimpleHTTPServer`.
+[/tip]
+
+You should be able to view the landing page for Lyrical Lightning, the Mobile Music Magic festival. 
+It has one link on the homepage to view the schedule and which stage the bands are on. 
+
+{{ image('	../../../../../static/img/docs/tutorials/tut-lyricallyghtning.png', 582, 557, alt='Image of PWA' ) }}
+
+Users of our site may have spotty network connectivity at the event when they'll likely want to access the schedule. 
+This makes a great candidate to turn it into a PWA that can be installed to our user's home screen, and provides all critical functionality even when offline. 
+
+
+# Create a Web App Manifest
+
+The [web app manifest ](https://developers.google.com/web/fundamentals/web-app-manifest/)is a simple JSON file that tells the browser about your web application 
+and how it should behave when 'installed' on the user's mobile device or desktop. Having a manifest is required by many browsers to show the 
+[Add to Home Screen prompt](https://developers.google.com/web/fundamentals/app-install-banners/).
+
+Add a file titled `manifest.json` to your repository with the following code:
+
+
+[sourcecode:JSON]
+{
+ "short_name": "LyLy",
+ "name": "Lyrical Lyghtning",
+ "icons": [
+   {
+     "src": "./images/amplogo192.png",
+     "type": "image/png",
+     "sizes": "192x192"
+   },
+   {
+     "src": "./images/amplogo512.png",
+     "type": "image/png",
+     "sizes": "512x512"
+   }
+ ],
+ "start_url": "/index.html",
+ "background_color": "#222325",
+ "display": "standalone",
+ "scope": "/",
+ "theme_color": "#222325"
+}
+[/sourcecode]
+
+
+# Add the AMP Service Worker
+
+A service worker is a script that your browser runs in the background, 
+separate from a web page, that extends the browsers features by caching requests to improve performance and providing offline functionality. 
+Building a service worker from scratch is possible but time consuming. 
+Libraries like Workbox help, but AMP goes one step further by offering the [AMP Service Worke](https://github.com/ampproject/amp-sw)r, 
+in which AMP automates a lot of steps directly, including the caching of AMP Scripts, assets and documents.
```

Comment:
Suggestion add somewhere: ".. and implements common best practices such as [navigation preload](https://developers.google.com/web/updates/2017/02/navigation-preload)"